GitCommands/EnvironmentPathsProvider.cs

```diff
@@ -73,22 +73,32 @@ public IEnumerable<string> GetEnvironmentValidPaths()
         // TODO: optimise?
         internal static bool IsValidPath(string path)
         {
-            FileInfo? fi = null;
             try
             {
-                fi = new FileInfo(path);
+                // NOTE:
+                // Path APIs don't throw an exception for invalid characters
+                // https://docs.microsoft.com/dotnet/core/compatibility/2.1#path-apis-dont-throw-an-exception-for-invalid-characters
+
+                _ = new FileInfo(path).Attributes;
+
+                return true;
             }
             catch (ArgumentException)
             {
             }
-            catch (PathTooLongException)
+            catch (IOException)
             {
+                // Querying attribures for UNC paths results in IOException
```

Comment:
💭 It seems like we could use a `path.StartsWith` check to avoid relying on exceptions for flow control, but I'm not sure it matters.